server/routes/opensearch.ts

```diff
@@ -177,10 +178,45 @@ export default class OpenSearchService {
     request: OpenSearchDashboardsRequest,
     opensearchDashboardsResponse: OpenSearchDashboardsResponseFactory
   ): Promise<IOpenSearchDashboardsResponse<any>> => {
+    let { isResultIndex } = request.params as { isResultIndex: any };
+    isResultIndex = JSON.parse(isResultIndex) as boolean;
+    let anomalyResultIndexMappings = {};
+
+    // If creating a custom result index: need to fetch the latest anomaly result index mappings.
+    // Currently pulling the stored JSON mappings from backend source code. Prefer to
+    // pull from backend rather than persisting a copy of the latest mappings in frontend code.
+    if (isResultIndex) {
+      try {
+        await fetch(ANOMALY_RESULTS_INDEX_MAPPINGS_LINK)
```

Comment:
Not sure if this will bring any security/performance issue, but seems AD plugin will be impacted if Github crashes and AD will run slow if Github has performance issue. Maybe we can either copy the mapping to frontend package or create a new API in backend if don't want to maintain duplicate index mappings in both frontend&backend package?

Reply:
Yeah still playing around with the correct approach here. In fact, I just realized this would break in future versions, since the link is tied to `main`. If we change the mappings in 1.4 for example, but are running Dashboards 1.3, then it would be expecting the mappings associated with 1.3, but would get the ones from 1.4, since it's pulling from `main`.

We would have a similar problem with the API approach, unless we added a param to specify the schema version - but that still doesn't solve the problem of needing to make frontend changes every time the result index schema is changed (e.g., would need to change the schema version in api call). 

Also, I think it's preferred to not add APIs if only for this very small use-case, so to me it seems the best approach could be persisting a copy in the frontend. That way the correct mappings can be coupled with the running version of the cluster. But even this may not be maintainable, and could be forgotten to be updated if more backend schema changes happen.

What do you think? I can sync with UX again to possibly remove this functionality so we don't risk it breaking in the future.

Reply:
I think another option is to just show some warning message like "Your detector is using custom result index: xxx, but it's not found in cluster. The index will be re-created when you start realtime job or historical analysis.".
So we don't provide re-create result index button. If user don't need to restart realtime/historical analysis, they actually don't need the custom result index. If they need to restart, AD backend will help recreate before starting realtime/historical job.

Reply:
Agreed, wording along the lines of that would be more clear and allow us to remove the create index button. Regardless I think it's agreed we don't want to persist mappings logic in the frontend at all.

Reply:
@ylwu-amzn UX has agreed to go with this option. I'll remove the create index button and just tune the wording in the callout.

Reply:
Thanks @ohltyler , will review once new commit published.

Reply:
New changes have been pushed.